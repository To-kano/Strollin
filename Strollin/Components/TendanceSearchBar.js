import React, { useState } from 'react';
import { connect } from 'react-redux';
import { StyleSheet, TextInput, View } from 'react-native';

import ButtonIcon from './ButtonIcon.js';

function SearchBar(props) {
  const [research, setresearch] = useState('');
  const { imagePath } = props;
  return (
    <View style={styles.container}>
      <View style={styles.horizontalDisplay}>
        <View style={styles.box}>
          <TextInput
            style={styles.textInput}
            placeholder="Search with tag"
            onChangeText={(text) => setresearch(text)}
            value={research}
          />
        </View>
        <ButtonIcon
          icon={require('../images/search.png')}
          onPress={() => {
            props.onPress(research);
            setresearch('');
          }}
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'flex-start',
    padding: 10,
  },
  box: {
    flex: 1,
  },
  horizontalDisplay: {
    width: '1000%',
    flexDirection: 'row',
    alignItems: 'center',
  },
  textInput: {
    height: 40,
    borderColor: 'gray',
    borderWidth: 1,
    backgroundColor: 'white',
    marginRight: 10
  },
});

const mapStateToProps = (state) => state;
export default connect(mapStateToProps)(SearchBar);
