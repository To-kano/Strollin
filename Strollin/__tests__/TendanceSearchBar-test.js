import React from 'react';
import {shallow} from 'enzyme';
import TendanceSearchBar from '../Components/TendanceSearchBar';
import ButtonIcon from '../Components/ButtonIcon';

import renderer from 'react-test-renderer';

import { StyleSheet, TextInput, View } from 'react-native';



const onPresstest = (test) => jest.fn();


describe('TendanceSearchBar', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const component = shallow(<TendanceSearchBar onPress={onPresstest}/>)
            expect(component).toMatchSnapshot()
        });
    });

    test('test render TendanceSearchBar', () => {
        renderer.create(
            <TendanceSearchBar onPress={onPresstest}/>,
        )      
      })

      it('handles button', () => {

        const wrapper =  shallow(
                <TendanceSearchBar onPress={onPresstest}/>
        );

        expect(wrapper.find(ButtonIcon).length).toBe(1);

        wrapper
        .find(ButtonIcon)
        .at(0)
        .props()
        .onPress()

        expect(wrapper.find(TextInput).length).toBe(1);

        wrapper
        .find(TextInput)
        .at(0)
        .props()
        .onChangeText()
  
      });
    
});