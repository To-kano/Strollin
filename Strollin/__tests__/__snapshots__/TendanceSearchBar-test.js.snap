// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TendanceSearchBar Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
